File: TripKitAndroid/src/main/java/com/skedgo/tripkit/TripUpdater.java
package com.skedgo.tripkit;

import androidx.annotation.NonNull;

import skedgo.tripkit.routing.Trip;

import io.reactivex.Observable;

public interface TripUpdater {
  @NonNull Observable<Trip> getUpdateAsync(@NonNull Trip trip);
  @NonNull Observable<Trip> getUpdateAsync(@NonNull String tripUrl);
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/TripUpdater.kt
package com.skedgo.tripkit

import com.skedgo.tripkit.routing.Trip
import io.reactivex.Observable
import kotlinx.coroutines.flow.Flow

interface TripUpdater {
    fun getUpdateAsync(trip: Trip): Observable<Trip>
    fun getUpdateAsync(tripUrl: String): Observable<Trip>
    fun getUpdateAsyncFlow(tripUrl: String): Flow<Trip>
    fun getUpdateAsyncFlow(trip: Trip): Flow<Trip>
}